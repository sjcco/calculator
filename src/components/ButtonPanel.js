import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = ({ clickHandler }) => {
  const handleClick = btnName => clickHandler(btnName);
  return (
    <div className="button-panel skin">
      <div className="btn-group">
        <Button wide={false} btnColor="bg-gray" name="AC" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-gray" name="+/-" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-gray" name="%" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-orange" name="รท" clickHandler={handleClick} />
      </div>
      <div className="btn-group">
        <Button wide={false} btnColor="bg-gray" name="7" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-gray" name="8" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-gray" name="9" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-orange" name="x" clickHandler={handleClick} />
      </div>
      <div className="btn-group">
        <Button wide={false} btnColor="bg-gray" name="4" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-gray" name="5" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-gray" name="6" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-orange" name="-" clickHandler={handleClick} />
      </div>
      <div className="btn-group">
        <Button wide={false} btnColor="bg-gray" name="1" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-gray" name="2" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-gray" name="3" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-orange" name="+" clickHandler={handleClick} />
      </div>
      <div className="btn-group">
        <Button wide btnColor="bg-gray" name="0" clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-gray" name="." clickHandler={handleClick} />
        <Button wide={false} btnColor="bg-orange" name="=" clickHandler={handleClick} />
      </div>
    </div>
  );
};

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default ButtonPanel;
