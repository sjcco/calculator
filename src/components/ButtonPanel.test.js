import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from './ButtonPanel';

test('render page correctly', () => {
  const mockHandler = jest.fn();
  const component = renderer.create(<ButtonPanel clickHandler={mockHandler} />).toJSON();
  expect(component).toMatchSnapshot();
});
