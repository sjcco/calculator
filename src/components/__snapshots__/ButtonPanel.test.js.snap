// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render page correctly 1`] = `
<div
  className="button-panel skin"
>
  <div
    className="btn-group"
  >
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="button bg-orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="btn-group"
  >
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="button bg-orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="btn-group"
  >
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="button bg-orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="btn-group"
  >
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="button bg-orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="btn-group"
  >
    <button
      className="button bg-gray zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button bg-gray"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="button bg-orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
